### PR TITLE
add types for snowflake-regex

### PR DESCRIPTION
**Pull request by ryanArora**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## snowflake-regex ([<kbd>unpkg</kbd>](https://unpkg.com/browse/snowflake-regex@latest/))
was missing the following properties:
1. dotAll
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 434a0e2e92b062179198235c30c1764175cda9cf
</p>

File: types/snowflake-regex/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": ["es6"],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "esModuleInterop": true,
+        "baseUrl": "../",
```
Comment (peterblazejewicz) [inline types/snowflake-regex/tsconfig.json:10]:
```suggestion
        "baseUrl": "../",
```
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

File: types/snowflake-regex/snowflake-regex-tests.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import regex, { generate } from 'snowflake-regex';
```
Comment (peterblazejewicz) [inline types/snowflake-regex/snowflake-regex-tests.ts:1]:
```suggestion
import regex = require('snowflake-regex');
import { generate } from 'snowflake-regex';
```

**Comment by ryanArora** [comment]
Sorry for the confusion about the Edits Infrastructure label, I committed an accidental change to package.json and reverted it.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ryanArora thx!

**Comment by ryanArora** [comment]
Ready to merge

**Pull request merged**